packages/eas-cli/src/build/android/prepareJob.ts

```diff
@@ -68,7 +69,7 @@ export async function prepareJobAsync(
       ...buildCredentials,
     },
     releaseChannel: ctx.buildProfile.releaseChannel,
-    updates: { channel: ctx.buildProfile.channel },
+    updates: { channel: resolveChannel(ctx) },
```

Comment:
If we are defining a default value for channel in cli we could also define a default for classic updates release channel, to unify the behavior, 

e.g.
```
...(isEASUpdateConfigured(...) 
  ?  { updates: { channel: resolveChannel(ctx) }} 
  :  { releaseChannel: ctx.buildProfile.releaseChannel ?? 'default' }
)
```

cc @brentvatne @jonsamp 
This PR is a breaking change for EAS Update, at the current state bare project that has release channel defined in native code and managed projects that have a config plugin that configures the release channel will be affected.

My suggestion above makes that situation even worse because in that case, the breaking changes would happen on classic updates too. Given that I'm not really suggesting doing that, just presenting this as an option.

Reply:
I don't have an opinion on this. I'm leaving the decision to @brentvatne and @jonsamp.

Reply:
Also, one other side effect of that change is that in the past we have validation in joi schema to allow only releaseChannel or updates.channel. In the past, it was fine because the user would have to define both of those fields in eas.json to have that conflict, but now this collision can happen if the release channel is defined and EAS Update is configured with the default channel. I'm not saying this is wrong behavior, but users might be confused by the error message.

We could either add some validation (if EAS Update is configured and profile.releaseChannel is defined throw descriptive error) or remove validation from a schema that enforces that only one of those values is set and let implementation define which value takes precedence.

Reply:
> We could either add some validation (if EAS Update is configured and profile.releaseChannel is defined throw descriptive error) or remove validation from a schema that enforces that only one of those values is set and let implementation define which value takes precedence.

It is already already implemented - https://github.com/expo/eas-cli/pull/1397

Reply:
I'd like to clarify the decision being made. What exactly are we choosing between?

In general
- we should not change any Classic Updates behavior
- if a user takes some action, like defining a channel in native code or running a config plugin that changes the channel, then we should use their channel instead of our default. We should only use our default channel name when there's (1) no channel defined in eas.json and (2) there's no channel written by a config plugin or specified in native code.